src/cli/publish_command.zig

```diff
@@ -451,9 +451,29 @@ pub const PublishCommand = struct {
         NeedAuth,
     };
 
+    fn isRepublishError(status_code: u32, response_body: []const u8) bool {
+        // Typical republish errors:
+        // - 403 Forbidden with message about already published version
+        // - 409 Conflict for some registries
+        if (status_code != 403 and status_code != 409) {
+            return false;
+        }
+
+        // Check for common republish error messages (case insensitive by checking lowercase versions)
+        return strings.containsComptime(response_body, "cannot publish over") or
+            strings.containsComptime(response_body, "already exists") or
+            strings.containsComptime(response_body, "already published") or
+            strings.containsComptime(response_body, "previously published") or
+            strings.containsComptime(response_body, "version already exists") or
+            strings.containsComptime(response_body, "Cannot publish over") or
+            strings.containsComptime(response_body, "Already exists") or
+            strings.containsComptime(response_body, "You cannot publish");
```

Comment:
```suggestion
```

I don't know all the messages, but this one seems like it could be used in other situations. Is it needed?